### PR TITLE
Layout for the field repeatable

### PR DESCRIPTION
**Pull request by Fedik**
Layout for the field repeatable, for j3.5

**test**
add repeatable field, example in some module options, 
and make sure that after patch this field works as before patch

example field XML:

``` xml
<field name="test" type="repeatable" icon="list"  label="Test" description="Test descr"
        maximum="5">
    <fieldset hidden="true" repeat="true">
        <field name="test_color" type="color" default="#eee"
            label="color"  description="test_color" />

        <field name="test_list" type="list" default="1"
            label="test_list" description="test_list" >
            <option value="1">JYES</option>
            <option value="0">JNO</option>
        </field>

        <field name="test_text" type="text"
            label="test_text" description="test_text" />

        <field name="test_textarea" type="textarea"
            label="textarea"  width="100" />

        <field name="test_radio" default="0" type="radio"
            label="test_radio" description="test_radio">
            <option value="1">JYES</option>
            <option value="0">JNO</option>
        </field>

        <field name="test_checkbox1" type="checkbox"
            label="checkbox1" description="test_checkbox1"  />

        <field name="test_checkboxes" type="checkboxes"
            label="checkboxes" description="test_checkboxes" >
            <option value="a">A</option>
            <option value="b">B</option>
            <option value="c">C</option>
            <option value="d">D</option>
        </field>
    </fieldset>
</field>
```

p.s.: I need make this pull against 3.5-dev branch, or?

### CODE REVIEW

File: layouts/libraries/joomla/form/fields/repeatable.php [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+$str[] = '</div>';
+
+// Button for display the modal window
+$select = JText::_($select);
+$icon   = $icon ? '<i class="icon-' . $icon . '"></i> ' : '';
+$str[]  = '<button class="open-modal btn" id="' . $id . '_button" >' . $icon . $select . '</button>';
+
+// Add scripts
+JHtml::_('bootstrap.framework');
+JHtml::_('script', 'system/repeatable.js', true, true);
```
Comment (dgt41) [inline layouts/libraries/joomla/form/fields/repeatable.php]: The first `true` is to load the framework, so this needs to be `false` or mootools gets loaded 
https://github.com/joomla/joomla-cms/blob/staging/libraries/cms/html/html.php#L672

Comment (Fedik) [inline layouts/libraries/joomla/form/fields/repeatable.php]: thanks, I will check ... I just copied it from the field file :wink:

**Comment by zero-24** [comment]
@Fedik How can we test this? Can you add some test instructions?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6917">issues.joomla.org/joomla-cms/6917</a>.</sub>

**Comment by Fedik** [comment]
@zero-24 added :wink:

**Comment by Devportobello** [comment]
@test Success
Tested with custom module, see to install: https://github.com/Devportobello/joomla-attach-PR-6917

**Comment by dgt41** [comment]
@test ok!
Can I make one more suggestion?
Since you use JHtml for the script changing line 83 to:
`JHtml::_('script', 'system/repeatable.min.js', false, true, false, true, true);`
and creating a `repeatable.min.js` as a compress version of `repeatable.js` we 'll save some bytes over the wire!

**Comment by Fedik** [comment]
I think better keep default behavior,
About `repeatable.min.js` there all files that in `media/system/js` have format similar to  `repeatable.js` and `repeatable-uncompressed.js` so here better stay as is .... I think it is another issue

@phproberto can you please check whether here is the right place for the layout file? currently `'libraries.joomla.form.fields.repeatable'`

**Comment by zero-24** [comment]
>repeatable-uncompressed.js has also to stay for B/C, I think (?)

IIRC the `uncompressed` is used with debug enabled.

**Comment by Devportobello** [comment]
Want to point out, [here](https://github.com/joomla/joomla-cms/pull/6858) i use another "way" to "store" the layout field, so what are the right "way" ?

**Comment by Fedik** [comment]
@Devportobello I think should be similar path as for the field file,
in case with calendar: the field in `libraries/joomla/form/fields/calendar.php` 
so the layout need place in that way: `libraries.joomla.form.fields.calendar`

@phproberto know more, will wait his feedback

**Comment by dgt41** [comment]
@zero-24 This needs a milestone 3.5
@Fedik @Devportobello Both you PRs as well as few more I did (#5654 #5655 #5871) actually have to wait for the new Layout renderer. There is #3231 and a comment that maybe https://github.com/joomla-projects/cms-naked/blob/master/libraries/cms/renderer/jlayout.php this is what will be used is all we have right now. Bottom line, most probably this line will have to be adjusted accordingly to the outcome of the layout render decision!

**Comment by Fedik** [comment]
@dgt41 thank for comment, I totally forgot that part :smile:

**Comment by designbengel** [comment]
tested like described with the testmodule - successfull<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6917">issues.joomla.org/joomla-cms/6917</a>.</sub>

**Comment by zero-24** [comment]
RTC Thanks!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6917">issues.joomla.org/joomla-cms/6917</a>.</sub>

**Comment by roland-d** [comment]
@Fedik isn't this pull request superseded by #7829 ?

**Comment by Fedik** [comment]
hm, yes and no :smile: 
I think PLT should decide ... whether we do the "layouts" for possible deprecated field or not :wink:

**Comment by roland-d** [comment]
I don't think we should update a deprecated field, gives the wrong signal to people. Better use the new and improved one :) If you agree, let's close this in favour of #7829

**Comment by Fedik** [comment]
you are right :wink:
I close here

**Pull request closed without merging**